classes/ActionScheduler_LogEntry.php

```diff
@@ -4,12 +4,37 @@
  * Class ActionScheduler_LogEntry
  */
 class ActionScheduler_LogEntry {
-	protected $action_id = '';
-	protected $message = '';
+	protected $action_id =  '';
+	protected $message =  '';
+	protected $date;
 
-	public function __construct( $action_id, $message ) {
+	/**
+	 * Constructor
+	 *
+	 * @param mixed  $action_id	Action ID
+	 * @param string $message   Message
+	 * @param string $date
+	 */
+	public function __construct( $action_id, $message, $date  = '' ) {
 		$this->action_id = $action_id;
-		$this->message = $message;
+		$this->message   = $message;
+		$this->date      = $date;
+	}
+
+	/**
+	 * Returns the date when this log entry was created
+	 *
+	 * The date is returned in a string, and there is a default format but it can be changed
+	 * through the `action_scheduler_date_format` filter.
+	 *
+	 * The date is the same timezone as the WordPress site.
+	 *
+	 * @return string
+	 */
+	public function get_date() {
+		$date = as_get_datetime_object( $this->date );
+		$date->setTimezone( ActionScheduler_TimezoneHelper::get_local_timezone() );
+		return $date->format( apply_filters( 'action_scheduler_date_format', 'Y-m-d H:i:s' ) );
```

Comment:
If the above suggestion is followed, and `$this->date` stores a `DateTime` object, then this can simply return the `DateTime` without messing with timezones or formatting and let the calling code format it however it needs.

Reply:
If you do want to provide a way to get the formatted value of the date, then we should add a new function, like `get_date_to_display()` or similar. But I wouldn't bother with that. Keep `ActionScheduler_LogEntry` super simple for now. 🙂